File: gamming/7/snake/spot.py
class Spot(object):

    def __init__(self, _x, _y, _w, _h):
        self.x = _x
        self.y = _y
        self.w = _w
        self.h = _h
        self.neighbors = []
        for i in range(4):
            self.neighbors.append(0)

    def show(self, _color):
        noStroke()
        fill(_color)
        rect(self.x*width/20, self.y*height/20, self.w, self.h)

    def addNeighbors(self, _grid):
        #UP
        if (self.y > 0):
            self.neighbors[0] = _grid[self.x][self.y-1]
        # RIGHT
        if (self.x < 19):    # Total 20x20, so the max id is 19
            self.neighbors[1] = _grid[self.x+1][self.y]
        # DOWN
        if (self.y < 19):    # Total 20x20, so the max id is 19
            self.neighbors[2] = _grid[self.x][self.y+1]
        # LEFT
        if (self.x > 0):
            self.neighbors[3] = _grid[self.x-1][self.y]

        if (self.y == 0):
            self.neighbors[0] = _grid[self.x][19]
        if (self.x == 19):
            self.neighbors[1] = _grid[0][self.y]
        if (self.y == 19):
            self.neighbors[2] = _grid[self.x][0]
        if (self.x == 0):
            self.neighbors[3] = _grid[19][self.y]

File: docs/public/interactive/mazeGenerator/Spot.js
class Spot {

	constructor(_i, _j) {
		this.i = _i;
		this.j = _j;
		this.w = width / (cols + 1);
		this.h = height / (rows + 1);
		this.x = (_i + 1) * this.w + this.w / 2;
		this.y = (_j + 1) * this.h + this.h / 2;
		this.visited = false;
		this.isWall = true;
		this.neighbors = [];
		this.gCost = Infinity;
		this.hCost = 0;
		this.fCost = 0;
		this.previous = null;
	}

	display(_color) {
		if (this.isWall) {
			fill('#AAAAAA');
			stroke('#AAAAAA');
		}
		else {
			stroke(_color);
			fill(_color);
		}
		rect(this.x, this.y, this.w, this.h);
		fill(0);
		noStroke();
		// if (openSet.includes(this) || closedSet.includes(this)) {
		// 	textSize(10);
		// 	text("G:" + nfc(this.gCost, 0), this.x - this.w / 3, this.y - this.h / 3);
		// 	text("H:" + nfc(this.hCost, 1), this.x + this.w / 3, this.y - this.h / 3);
		// 	textSize(10);
		// 	text("F:" + nfc(this.fCost, 1), this.x, this.y + this.h / 3);
		// }
	}

	highLight() {
		stroke('#FF0000');
		fill('#FF0000');
		rect(this.x, this.y, this.w, this.h);
	}

	addNeighbors(spots) {
		if (this.isWall) return;
		const { i, j } = this;
		if (i > 0 && !spots[i - 1][j].isWall) this.neighbors.push(spots[i - 1][j]);
		if (i < cols - 1 && !spots[i + 1][j].isWall) this.neighbors.push(spots[i + 1][j]);
		if (j > 0 && !spots[i][j - 1].isWall) this.neighbors.push(spots[i][j - 1]);
		if (j < rows - 1 && !spots[i][j + 1].isWall) this.neighbors.push(spots[i][j + 1]);


	}

	Visited() {
		this.visited = true;
		this.isWall = false;
	}

	isVisited() {
		return this.visited;
	}

	isWall() {
		return this.isWall;
	}


}
